websites/Q/Quran.com/presence.ts

```diff
@@ -0,0 +1,86 @@
+const presence = new Presence({
+		clientId: "969064871310282813"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+function getElementByXpath(path: string) {
+	return document.evaluate(
+		path,
+		document,
+		null,
+		XPathResult.FIRST_ORDERED_NODE_TYPE,
+		null
+	).singleNodeValue;
+}
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "logo_1024",
+			startTimestamp: browsingTimestamp
+		},
+		{ pathname } = window.location;
+	switch (pathname) {
+		case "/": {
+			presenceData.details = "Browsing the homepage...";
+			break;
+		}
+		case "/radio": {
+			presenceData.details = "Looking through radio stations...";
+			break;
+		}
+		case "/about-us": {
+			presenceData.details = "Viewing the about us page";
+			break;
+		}
+		case "/apps": {
+			presenceData.details = "Looking at Quran apps";
+			break;
+		}
+		case "/developers": {
+			presenceData.details = "Looking at the developers page";
+			break;
+		}
+		case "/privacy": {
+			presenceData.details = "Reading privacy terms";
+			break;
+		}
+		case "/support": {
+			presenceData.details = "Looking at the support page";
+			break;
+		}
+		case "/search": {
+			if (document.title.includes("Search for")) {
+				presenceData.details = `Making a ${document.title
+					.split("-")[0]
+					.toLowerCase()}`;
+			} else presenceData.details = "Searching for something...";
+			break;
+		}
+		default: {
+			if (pathname.includes("/reciters")) {
+				if (pathname.includes("/reciters/")) {
+					presenceData.details = `Viewing a reciter: ${
+						document.querySelector<HTMLDivElement>(
+							"#__next > div > div.index_pageContainer__Pxtn3 > div.reciterPage_reciterInfoContainer__mYOjC > div > div > div:nth-child(2) > div.ReciterInfo_reciterName__SiK59"
+						).textContent
+					}`;
+				} else presenceData.details = "Looking for a reciter";
+			} else {
+				presenceData.details = "Reading the Holy Quran";
+				presenceData.state = `Surah: Surat ${
+					getElementByXpath("/html/body/div[1]/div/div[2]/div/div[1]/div/p")
+						.textContent
+				} (${
+					getElementByXpath(
+						"/html/body/div[1]/div/div[2]/div/div[2]/div/p[2]/span[2]"
+					)
+						? getElementByXpath(
+								"/html/body/div[1]/div/div[2]/div/div[2]/div/p[2]/span[2]"
+						  ).textContent
+						: getElementByXpath(
+								"/html/body/div[1]/div/div[2]/div/div[2]/div/p[2]/span"
+						  ).textContent
```

Comment:
use optional chaining and nullish coalescing

Reply:
![image](https://user-images.githubusercontent.com/96930330/172503737-7cb68563-32b3-4abd-ad98-79b96f4c419b.png)
If you got a way to do it, I'm all ears; in the meantime I'm stuck with this method. Will keep unresolved for your response (hopefully there's one that satisfies you)

Reply:
It really isn't that hard
```suggestion
					getElementByXpath(
						"/html/body/div[1]/div/div[2]/div/div[2]/div/p[2]/span[2]"
					) ??
					getElementByXpath(
								"/html/body/div[1]/div/div[2]/div/div[2]/div/p[2]/span"
						  ).textContent
```